### DISCUSSION TITLE
How to apply a postprocessor to a specific actor

### DISCUSSION THREAD

**Discussion by Ziphil** [excaliburjs/Excalibur#2492] [Q&A]
Hello,
First thank you so much for this great engine! I'm really enjoying building a game with it.

I've read [this documentation](https://excaliburjs.com/docs/postprocessors) and tried to apply a postprocessor to a specific actor. I've tried the following but ended up seeing the postprocessor applied to the whole screen:
```
const postProcessor = /* instanciate a postprocessor */
actor.graphics.onPreDraw = (context) => {
  context.save()
  context.addPostProcessor(postProcessor)
  context.restore()
}
```

Is there a way to apply a postprocessor (or a WebGL fragment shader) to only a specific actor?

**Comment by eonarheim** [comment]
Hi @Ziphil thanks for the kind words!

At the moment we don't have a way for postprocessors to apply to a specific actor. 

This is something I've been thinking about though, potentially applied to Raster graphics like the `ex.Canvas` and/or prehaps to `ex.ImageSource`. 

Let me know what you think!

**Comment by pixelprodukt** [comment]
I would also vote for the possibility to add a shader to specific actors, that's the one thing I'm currently missing. Kinda to do like outlines on interactable objects or 'shadow'-vision if some actor is behind something and so on.

**Reply by eonarheim** [reply]
@pixelprodukt I've been thinking about this more, I wonder if we might be able to have a custom renderer attached to specific actors, which would give you the specific shader.

**Reply by pixelprodukt** [reply]
You mean like a custom renderer component? That would be nice I think.

**Reply by eonarheim** [reply]
@pixelprodukt @Ziphil We've merged per actor shaders into alpha! Check them out https://github.com/excaliburjs/Excalibur/pull/2670

This will land in v0.28.0 very soon, but you can try the latest alpha bits
